backend/Recodo.API/Controllers/UserController.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Recodo.BLL.Services;
+using Recodo.Common.Dtos.User;
+using System.Threading.Tasks;
+
+namespace Recodo.API.Controllers
+{
+    [ApiController]
+    [Route("api/User")]
+    public class UserController : ControllerBase
+    {
+        private readonly AuthService _authService;
+        private readonly UserService _userService;
+
+        public UserController(AuthService authService, UserService userService)
+        {
+            _authService = authService;
+            _userService = userService;
+        }
+
+        [HttpPost("Update")]
+        public async Task<IActionResult> Update([FromForm] UpdateUserDTO userDTO, [FromForm] IFormFile avatar)
+        {
+            await _userService.UpdateUser(userDTO, avatar);
+            return NoContent();
+        }
+
+        [HttpPost("UpdatePassword")]
+        public async Task<IActionResult> UpdatePassword([FromBody] UpdateUserDTO userDTO)
+        {
+            await _userService.UpdateUserPassword(userDTO);
+            return NoContent();
+        }
+
+        [HttpPost("ResetPassword")]
+        public async Task<IActionResult> ResetPassword([FromBody] UpdateUserDTO userDTO)
```

Comment:
It should be enough to just have the user's id, old and new passwords to make this action. No need to get all their info